### PR TITLE
check if libimobiledevice executables exist

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

We've been noticing several ArgumentErrors in crash logs from missing libimobiledevice artifacts. In place of a full solution (checksums, cache verification), we can add a check that the specific artifacts exist on disk to the cache update logic.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/cache_test.dart [hunk 1]
```diff
@@ -330,13 +330,45 @@ void main() {
...
 
-  group('Unsigned mac artifacts', () {
+  group('macOS artifacts', () {
     MockCache mockCache;
 
     setUp(() {
       mockCache = MockCache();
     });
 
+    testUsingContext('verifies executables for libimobiledeivce in isUpToDateInner', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/cache_test.dart]: `libimobiledevice` spelling

**Review by christopherfujino — APPROVED** [review]
LGTM, other than spelling nit for test name

File: packages/flutter_tools/lib/src/cache.dart [hunk 2]
```diff
@@ -1102,18 +1102,42 @@ class IosUsbArtifacts extends CachedArtifact {
     'ios-deploy',
   ];
 
+  static const Map<String, List<String>> _kExecutables = <String, List<String>>{
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/cache.dart]: Oh, maybe a comment here too, to give some context

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/cache.dart]: Done

File: packages/flutter_tools/test/general.shard/cache_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/cache_test.dart]: Fixed

**Pull request merged**